### PR TITLE
[Buttons] Add color prop to buttons

### PR DESCRIPTION
**Pull request by dbushy727**
# Button Colors

TL DR; buttons can be different colors now. Just supply the `buttonColor` prop to your button and Voila!

`<Button buttonColor={COLORS.statusRed}>Click Me!</Button>`

### Check it out here --> [Review App](https://curology-radiance-pr-219.herokuapp.com/?path=/story/buttons--button)
<br>

large

<hr>

Based on the latest design for [Dupe Patients project](https://www.figma.com/file/S4CWI1jVFMxsJZWYDUaNn0/Duplicate-Patient-Support-Experience?node-id=245%3A3103) we have to render a red button. 

This PR adds the ability to supply a color prop to any of the buttons, and it will render the buttons in that color in all the various states. Please note that it is required to use one of the colors from one of the constants files. 

The story has been updated to allow a user to pick any color they like and view it. 

Certain colors are designated alternate colors for various states, while others do not. In this PR, we have linked some colors together and derive other colors where that link does not exist.

For example:
primary -> purple80 for the RoundButton background and text respectively.
statusRed -> lighter desaturated version via tinycolor
Screen Shot 2020-04-15 at 11 46 56 AM
Screen Shot 2020-04-15 at 11 47 18 AM
Screen Shot 2020-04-15 at 11 47 10 AM

Screen Shot 2020-04-15 at 11 51 06 AM

### CODE REVIEW

File: src/shared-components/button/constants.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { COLORS } from '../../constants';
+
+// TODO: potentially break out these pairings to the color constants file
+export const textColorsAssociatedWithColors = {
```
Comment (dbushy727) [inline src/shared-components/button/constants.js:4]: wasnt sure if we wanted this in our actual constants/colors file since its kinda pointed to how i was using it. Maybe there's a better way to connect colors together as related.

File: src/shared-components/button/components/roundButton/style.js [hunk 2]
```diff
@@ -62,6 +64,54 @@ export const roundButtonLoader = disabled => css`
...
   margin: 10px 0;
 `;
+
+/**
+ * Given a color as an argument,
+ * determine an alternate color for pairing
+ * @param  string color   the current color name of the round button (e.g purple, primary, etc.)
+ * @return string         hex string of the alternate color (e.g. #efefef)
+ */
+const determineAlternateTextColor = color => {
```
Comment (dbushy727) [inline src/shared-components/button/components/roundButton/style.js:78]: i didnt put this as a util as its kind of opinionated as to whats a derived alternate color. This works great for the round button color vs text color, but this may not be the formula we want across the board, so i left this as helper functions in the roundbutton styles specifically. Happy to break this out if we need.

**Comment by dbushy727** [comment]
> This looks good. I would add one small thing to the docs that just says we probably want to keep usage of these buttons to be special cases only (at least for the time being)

added a comment to the docs in 14b391ed6cf3e28466ab60847646f4c5011acb9f

`We should generally try to use the default button color when possible. Only for special cases should we need to use a different button color.`

File: src/shared-components/button/index.js [hunk 3]
```diff
@@ -44,6 +46,7 @@ class Button extends React.Component {
   static defaultProps = {
     disabled: false,
     buttonType: 'primary',
+    color: 'purple',
```
Comment (daigof) [inline src/shared-components/button/index.js:49]: ok so dont hate me for this but as I pointed out above I think it would be more appropriate if we pass a COLORS constant as prop value it is safer type-wise and make more consistent with our Radiance usage (for example Icons, callout, etc we pass the hex constant value or one of COLORS constants)

File: docs/button/button.md [hunk 4]
```diff
@@ -15,6 +15,20 @@ import { CheckmarkIcon } from 'radiance-ui/lib/icons';
     <Button disabled>Disabled Button</Button>
     <Button isFullWidth>Full Width Button</Button>
 
+    <Button color="statusRed">Primary Color</Button>
```
[the remark this answers is not in the document]
Comment (dbushy727) [reply docs/button/button.md:18]: @daigof the problem with doing ${COLORS.statusRed} is that it gives the value, not the key. I need the key name like "statusRed" not "#abc123"

[the remark this answers is not in the document]
Comment (dbushy727) [reply docs/button/button.md:18]: ok, i changed the implementation to accept the color value instead. it makes the knobs dropdown work a little weird, but otherwise everything else is fine.

I will say that it feels a bit uncommon to pass a hex argument from an enumerated list rather than key names which is typical of other ui frameworks (semantic, bootstrap, foundation, material, etc)

For the sake of this pr i will make use of the `buttonColor={COLORS.statusRed}` but would def like to talk about the pros and cons offline of this.

File: docs/button/linkButton.md [hunk 5]
```diff
@@ -27,7 +25,7 @@ import { LinkButton } from 'radiance-ui';
       I am Disabled
     </LinkButton>
   </LinkButton.Container>
-</React.Fragment>
+</React.Fragment>;
```
[the remark this answers is not in the document]
Comment (dbushy727) [reply docs/button/linkButton.md:28]: fixed

File: src/shared-components/button/index.js (hunk 3 above)
Comment (dbushy727) [reply src/shared-components/button/index.js]: haha no dont worry, i dont hate you 😄 i get the idea of the switch over. I made the change for now but as mentioned above, would love to talk more about this for future stuff.

**Comment by dbushy727** [comment]
@daigof ok i believe i addressed all your concerns. The only thing that isnt 100% exactly is the story for the RoundButton. the layout (horizontal vs vertical) is different than the rest of the button pages, so i opted to conform to the current page style rather than make it appear like the other pages.

**Comment by Shemoel1** [comment]
100% agree with @benkolde on special case use. Looks good to me otherwise @dbushy727
